Source/com/Communicator.cpp

```diff
@@ -33,6 +33,82 @@ namespace RPC {
 
     static Core::ProxyPoolType<RPC::AnnounceMessage> AnnounceMessageFactory(2);
 
+    class DynamicLoaderPaths {
+    private:
+        static constexpr TCHAR LoaderConfig[] = _T("/etc/ld.so.conf");
+
+    public:
+        DynamicLoaderPaths(const DynamicLoaderPaths&) = delete;
+        DynamicLoaderPaths& operator= (const DynamicLoaderPaths&) = delete;
+
+        DynamicLoaderPaths() 
+            : _downloadLists()
+        {
+            ReadList(LoaderConfig, _downloadLists);
+            _downloadLists.emplace_back(_T("/usr/lib/"));
+            _downloadLists.emplace_back(_T("/lib/"));
+        }
+        ~DynamicLoaderPaths() = default;
+
+    public:
+        const std::vector<string> Paths() const
```

Comment:
Lets make this a const std::vector<string>& here as a return value. Otherwise a copy is created...

Reply:
done